### ISSUE TITLE
Unable to open PDF files in browser

### ISSUE THREAD

**Issue by srirambv** [brave/browser-laptop#4335]
**Did you search for similar issues before submitting this one?**
Yes

**Describe the issue you encountered:**
Unable to open PDF files in browser

**Expected behavior:**
Should load pdf files in browser window

- Platform (Win7, 8, 10? macOS? Linux distro?):
Ubuntu 16.04

- Brave Version:
0.12.3dev-RC1 (Clean Install)

- Steps to reproduce:
    1. Unable to open the file http://www.orimi.com/pdf-test.pdf

- Screenshot if needed:
![image](https://cloud.githubusercontent.com/assets/17010094/18888700/5b6aac48-84f1-11e6-8c33-d524cc162d92.png)

- Any related issues:
#4329

**Comment by bbondy** [comment]
@diracdeltas does this look like an issue that we already solved in our forked version of pdfjs? We're using the Chromium one now, maybe we need to package our own extension with a different ID for pdfjs.

**Comment by bbondy** [comment]
@srirambv I think if you click a link it works but if you paste directly in the URL bar it doesn't. Can you confirm?
http://martinfowler.com/ieeeSoftware/whenType.pdf

**Comment by srirambv** [comment]
@bbondy Yes. Only works when you click on a pdf link. Opening the pdf link in a new tab or pasting the link directly in URL bar it doesn't work.

**Comment by diracdeltas** [comment]
@bbondy yes this is an issue i fixed in our fork; can we revert to using that for now?

**Comment by diracdeltas** [comment]
i also see this error in master, which may be related. it's due to an unsupported API, which is stubbed out in our fork.
```
[68211:0927/154121:INFO:CONSOLE(0)] "Error in response to extension.isAllowedFileSchem$Access: TypeError: Cannot read property 'onBeforeNavigate' of undefined
    at null.callback (chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/pdfHandler.j$:200:23)
    at chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/pdfHandler.js:191:18", sour$e: chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/pdfHandler.html (0)
[68211:0927/154121:INFO:CONSOLE(0)] "Unchecked runtime.lastError while running extensi$n.isAllowedFileSchemeAccess: Access to extension API denied.
    at chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/pdfHandler.js:191:18", sour$e: chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/pdfHandler.html (0)
[68211:0927/154121:INFO:CONSOLE(0)] "Unchecked runtime.lastError while running storage$get: "sync" is not available in this instance of Chrome
    at next (chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/telemetry.js:157:39)
    at getLoggingPref (chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/telemetry.j$:144:5)
    at maybeSendPing (chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/telemetry.js$49:5)
    at chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/telemetry.js:45:3
    at chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/telemetry.js:162:3", source$ chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/pdfHandler.html (0)
[68211:0927/154121:ERROR:extension_function_dispatcher.cc(613)] Unknown Extension API $ extension.isAllowedFileSchemeAccess
[68211:0927/154121:INFO:CONSOLE(0)] "Unchecked runtime.lastError while running storage.
get: "managed" is not available in this instance of Chrome
    at next (chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/telemetry.js:157:39)
    at null.callback (chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/telemetry.js:
158:9)", source: chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/pdfHandler.html (0
)
```

**Comment by bbondy** [comment]
Fixed for RC2 here:
https://github.com/brave/browser-laptop/commit/e611a31e0334ef63cfa11c84eba3eeb64e81f3a2

**Comment by bbondy** [comment]
we now maintain our own PDFJS fork instead of using the chrome extension unchanged.

**Comment by capouch** [comment]
Should opening PDF files work now in 0.12.6 (Linux Slackware)?  If it does, I haven't found the magic way to do it.  I have tried in regular tabs, private tabs, pasting in file:// URLs, enabling/disabling HTML5 PDF viewer, and no matter what it does the same thing: brings up a file save dialog. 

I did find a kludgy workaround: I bookmarked a URL that leads to a PDF, and once I've clicked it, I can use the "Open" button on the page to open my local file.  But nothing else, so far, works for me.
